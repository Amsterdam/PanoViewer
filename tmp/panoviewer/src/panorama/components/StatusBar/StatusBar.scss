@import "../../../shared/styles/config";

.c-panorama-status-bar {
  background-color: $primary-light-alpha67;
  bottom: 0;
  color: $primary-dark;
  height: $button-size;
  line-height: $button-size;
  padding-right: $base-whitespace / 2;
  position: absolute;
  width: 100%;
}

.c-panorama-status-bar__history {
  border-right: 1px solid $secondary-gray20;
  bottom: 0;
  left: 0;
  position: absolute;
  z-index: 2;
}

.c-panorama-status-bar__info {
  @include font-size($xs-font);
  @include font-weight($bold-weight);

  bottom: 0;
  height: $button-size;
  left: 202px;
  line-height: $button-size;
  overflow: hidden;
  position: absolute;
  right: 0;
}

.c-panorama-status-bar__info-item {
  border-left: solid $secondary-gray20 1px;
  float: right;
  padding: 0 $base-whitespace;
}

.c-panorama-status-bar-status-bar__history {
  border: none;
  padding-left: $base-whitespace;

  &::before {
    display: none;
  }

  &:hover,
  &:active {
    border: none;
  }
}

.c-panorama-status-bar-status-bar__coordinates {
  border: none;
}
