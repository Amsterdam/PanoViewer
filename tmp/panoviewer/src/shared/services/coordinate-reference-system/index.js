import { wgs84ToRd,
  rdToWgs84,
  parseLocationString,
  normalizeCoordinate,
  normalizeLocation } from './crs-converter';

export {
  wgs84ToRd,
  rdToWgs84,
  parseLocationString,
  normalizeCoordinate,
  normalizeLocation
};
