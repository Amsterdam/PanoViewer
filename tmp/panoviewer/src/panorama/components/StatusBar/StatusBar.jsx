import React from 'react';
import PropTypes from 'prop-types';
import './StatusBar.scss';
import { wgs84ToRd } from '../../../shared/services/coordinate-reference-system/crs-converter';
import { dateToString } from '../../../shared/services/date-formatter/date-formatter';
import PanoramaToggleContainer from '../PanoramaToggle/PanoramaToggleContainer';

const convertLocation = (location) => {
  const [latitude, longitude] = location;
  const { x: rdX, y: rdY } = wgs84ToRd({ latitude, longitude });
  const formattedWgs84Location = `${latitude.toFixed(7)}, ${longitude.toFixed(7)}`;

  return `${rdX.toFixed(2)}, ${rdY.toFixed(2)} (${formattedWgs84Location})`;
};

const StatusBar = ({ date, location, heading, currentLabel }) => (
  <div className="c-panorama-status-bar">
    <PanoramaToggleContainer {...{ heading, currentLabel, location }} />

    <div className="c-panorama-status-bar__info">
      <div className="c-panorama-status-bar__info-item">
        <span>{convertLocation(location)}</span>
      </div>
      <div className="c-panorama-status-bar__info-item">
        <span>{dateToString(date)}</span>
      </div>
    </div>
  </div>
);

StatusBar.defaultProps = {
  date: ''
};

StatusBar.propTypes = {
  heading: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  date: PropTypes.oneOfType([PropTypes.string, PropTypes.instanceOf(Date)]).isRequired,
  currentLabel: PropTypes.string.isRequired,
  location: PropTypes.array.isRequired //eslint-disable-line
};

export default StatusBar;
