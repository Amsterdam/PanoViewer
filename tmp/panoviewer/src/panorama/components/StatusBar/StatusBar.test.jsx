import React from 'react';
import { shallow } from 'enzyme';
import StatusBar from './StatusBar';

import { wgs84ToRd } from '../../../shared/services/coordinate-reference-system/crs-converter';

jest.mock('../../../shared/services/coordinate-reference-system/crs-converter');
jest.mock('../../../shared/services/date-formatter/date-formatter');

describe('StatusBar', () => {
  it('should render', () => {
    const props = {
      heading: 999,
      date: '',
      currentLabel: 'Meest recent',
      location: [2, 3]
    };

    wgs84ToRd.mockReturnValue(({ x: 12, y: 9 }));

    const component = shallow(
      <StatusBar {...props} />
    );
    expect(component).toMatchSnapshot();
  });
});
