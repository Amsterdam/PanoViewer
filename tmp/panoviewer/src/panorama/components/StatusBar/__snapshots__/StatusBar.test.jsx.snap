// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusBar should render 1`] = `
<div
  className="c-panorama-status-bar"
>
  <Connect(PanoramaToggle)
    currentLabel="Meest recent"
    heading={999}
    location={
      Array [
        2,
        3,
      ]
    }
  />
  <div
    className="c-panorama-status-bar__info"
  >
    <div
      className="c-panorama-status-bar__info-item"
    >
      <span>
        12.00, 9.00 (2.0000000, 3.0000000)
      </span>
    </div>
    <div
      className="c-panorama-status-bar__info-item"
    >
      <span />
    </div>
  </div>
</div>
`;
